ground/gcs/src/plugins/boards_brainfpv/brainre1.cpp

```diff
@@ -77,6 +77,22 @@ bool BrainRE1::queryCapabilities(BoardCapabilities capability)
     case BOARD_CAPABILITIES_OSD:
     case BOARD_CAPABILITIES_UPGRADEABLE:
         return true;
+    case BOARD_CAPABILITIES_MAGS:
+        {
+            ExtensionSystem::PluginManager *pm = ExtensionSystem::PluginManager::instance();
+            UAVObjectManager *uavoManager = pm->getObject<UAVObjectManager>();
+            HwBrainRE1 *hwBrainRE1 = HwBrainRE1::GetInstance(uavoManager);
+            Q_ASSERT(hwBrainRE1);
+
+            if (!hwBrainRE1)
+                return false;
+
+            HwBrainRE1::DataFields settings = hwBrainRE1->getData();
```

Comment:
You can just getField the one you want rather than everything. The upside of that method is less work later to move away from hard-coded/compiled UAVOs (assuming you also use the getObject method to get the object on the line above). Will accept either way so up to you.